packages/lodestar-cli/src/commands/beacon.ts

```diff
@@ -45,27 +60,51 @@ export class BeaconNodeCommand implements ICliCommand {
     generateCommanderOptions(command, BeaconNodeOptions);
   }
 
-  public async action(options: IBeaconCommandOptions, logger: ILogger): Promise<void> {
-    let conf: Partial<IBeaconNodeOptions> = {};
+  public async action(cmdOptions: IBeaconCommandOptions, logger: ILogger): Promise<void> {
+    let nodeOptions: Partial<IBeaconNodeOptions> = {};
+    //find better place for this once this cli is refactored
+    await initBLS();
 
-    if (options.loggingLevel) {
+    if (cmdOptions.loggingLevel) {
       // eslint-disable-next-line no-undef
       // @ts-ignore
-      logger.setLogLevel(LogLevel[options.loggingLevel]);
+      logger.setLogLevel(LogLevel[cmdOptions.loggingLevel]);
     }
 
     //merge config file
-    if (options.configFile) {
-      const parsedConfig = getTomlConfig(options.configFile, BeaconNodeOptions);
+    if (cmdOptions.configFile) {
+      const parsedConfig = getTomlConfig(cmdOptions.configFile, BeaconNodeOptions);
       //cli will override toml config options
-      conf = deepmerge(conf, parsedConfig) as Partial<IBeaconNodeOptions>;
+      nodeOptions = deepmerge(nodeOptions, parsedConfig) as Partial<IBeaconNodeOptions>;
     }
     //override current config with cli config
-    conf = deepmerge(conf, optionsToConfig(options, BeaconNodeOptions));
+    nodeOptions = deepmerge(nodeOptions, optionsToConfig(cmdOptions, BeaconNodeOptions));
+    const peerId = await getPeerId(cmdOptions.peerId);
+    const defaultDiscv5Opt = {
+      enr: ENR.createFromPeerId(peerId),
+      bindAddr: "/ip4/0.0.0.0/udp/5501",
+      bootEnrs: [] as ENR[]};
+    const discv5 = nodeOptions.network? Object.assign(defaultDiscv5Opt, nodeOptions.network.discv5) : defaultDiscv5Opt;
```

Comment:
Why isn't this config inline with other configurations?

Reply:
I think because we need the PeerId to create the local ENR if it isn't already specified.

Reply:
Aren't we suppose to modify enr in runtime according to subnets, how do we do that?

Reply:
the ENR gets passed into the network in the constructor. As the network subscribes / unsubscribes to subnets and fork versions change, the ENR can get updated. We should ideally be creating the whole network identity (peer Id and ENR) up front, and storing it to disk, or reading it from disk.